test/scripts/browser-test/run-test.mjs

```diff
@@ -0,0 +1,231 @@
+/**
+ * This script is served by `harness.mjs` and runs in the browser.
+ */
+import { WASI, File, OpenFile, ConsoleStdout, PreopenDirectory } from 'https://cdn.jsdelivr.net/npm/@bjorn3/browser_wasi_shim@0.3.0/+esm'
+import { polyfill } from 'https://cdn.jsdelivr.net/npm/wasm-imports-parser@1.0.4/polyfill.js/+esm';
+
+/**
+ * @param {{
+ *   module: WebAssembly.Module,
+ *   addToImports: (importObject: WebAssembly.Imports) => Promise<void>
+ * }}
+ */
+export async function instantiate({ module, addToImports }) {
+    const args = ["target.wasm"]
+    const env = []
+    const fds = [
+        new OpenFile(new File([])), // stdin
+        ConsoleStdout.lineBuffered((stdout) => {
+            console.log(stdout);
+        }),
+        ConsoleStdout.lineBuffered((stderr) => {
+            console.error(stderr);
+        }),
+        new PreopenDirectory("/", new Map()),
```

Comment:
Shouldn't there be some way to use the preopens we pass along in the `--dir` CLI args... or no?

How do the tests that need special access work?! Or maybe that's just the `fts.c` test` that we skip in browser?

Reply:
Good catch. The preopen is unnecessary and I don't think it's even worth adding `--dir` emulation support to the harness. File system emulation is not part of wasi-libc's responsibility and it's not the main focus here, so I just skipped tests using `--dir` for now.